unified-codes/tools/scripts/dgraph/migration/v1/export.js

```diff
@@ -0,0 +1,190 @@
+  if (descriptionA < descriptionB) {
+    return -1;
+  }
+  if (descriptionA > descriptionB) {
+    return 1;
+  }
+
+  const formCategoryA = itemA.formCategory;
+  const formCategoryB = itemB.formCategory;
+  if (formCategoryA < formCategoryB) {
+    return -1;
+  }
+  if (formCategoryA > formCategoryB) {
+    return 1;
+  }
+
+  const formA = itemA.form;
+  const formB = itemB.form;
+  if (formA < formB) {
+    return -1;
+  }
+  if (formA > formB) {
+    return 1;
+  }
+
+  return 0;
+};
+const writeFile = async () => {
+  const workbook = new excel.Workbook();
+  const worksheet = workbook.addWorksheet('Products');
+  let rowNum = 2;
+
+  worksheet.columns = [
+    { header: 'Product Code', key: 'product_code', width: 15 },
+    { header: 'Product', key: 'product_description', width: 32 },
+    { header: 'Type', key: 'product_type', width: 15 },
+    { header: 'DDD', key: 'product_ddd', width: 15 },
+    { header: 'RxNav', key: 'product_rxnav', width: 15 },
+    { header: 'WHO EML', key: 'product_eml', width: 15 },
+    { header: 'WHO ATC', key: 'product_atc', width: 15 },
+    { header: 'UNSPSC', key: 'product_unspsc', width: 15 },
+    { header: 'NZULM', key: 'product_nzulm', width: 15 },
+    { header: 'Form Category', key: 'form_category', width: 15 },
+    { header: 'Form', key: 'form', width: 15 },
+    { header: 'Item Code', key: 'item_code', width: 10 },
+    { header: 'Item', key: 'item_description', width: 50 },
+    { header: 'Strength', key: 'item_strength', width: 30 },
+    { header: 'DDD', key: 'item_ddd', width: 15 },
+    { header: 'RxNav', key: 'item_rxnav', width: 15 },
+    { header: 'WHO EML', key: 'item_eml', width: 15 },
+    { header: 'WHO ATC', key: 'item_atc', width: 15 },
+    { header: 'UNSPSC', key: 'item_unspsc', width: 15 },
+    { header: 'NZULM', key: 'item_nzulm', width: 15 },
+  ];
+  worksheet.getRow(1).font = { bold: true };
+
+  items.sort(sortItems).forEach((item, n) => {
+    const row = worksheet.getRow(rowNum++);
+    let cell = 1;
+    row.getCell(cell++).value = item.product.code;
+    row.getCell(cell++).value = item.product.description;
+    row.getCell(cell++).value = item.product.type;
+    row.getCell(cell++).value = item.product.ddd;
+    row.getCell(cell++).value = item.product.rxnav;
+    row.getCell(cell++).value = item.product.eml;
+    row.getCell(cell++).value = item.product.atc;
+    row.getCell(cell++).value = item.product.unspsc;
+    row.getCell(cell++).value = item.product.nzulm;
+    row.getCell(cell++).value = item.formCategory;
+    row.getCell(cell++).value = item.form;
+    row.getCell(cell++).value = item.code;
+    row.getCell(cell++).value = item.description;
+    row.getCell(cell++).value = item.strength;
+    row.getCell(cell++).value = item.ddd;
+    row.getCell(cell++).value = item.rxnav;
+    row.getCell(cell++).value = item.eml;
+    row.getCell(cell++).value = item.atc;
+    row.getCell(cell++).value = item.unspsc;
+    row.getCell(cell++).value = item.nzulm;
```

Comment:
If you made an array of, e.g. `["code", "description", ...]`, you could prob do this with a `forEach` function, e.g.

```
itemProperties.forEach((property) => row.getCell(cell++).value = item[property]);
```

Might be a bit more tricky with the `product.x` bits, but could nest that bit?